Remove unnecessary special handling of large fields

senier opened the pull request:
Close #531

I'm not sure why the special handling for unaligned fields / field larger than 7 bit even existed when using a bit array. All test still work, hence I assume removing that code should be OK. Please have a look.

rssen reviewed (commented):
I see no problem removing this.
We do already have a test `test_message_value_odd_length_binary` that is supposed to test the parsing of unaligned fields. Maybe this can be removed now, as your new test is more expressive.

senier commented:
> I see no problem removing this.
> We do already have a test `test_message_value_odd_length_binary` that is supposed to test the parsing of unaligned fields. Maybe this can be removed now, as your new test is more expressive.

Good catch - this test is not too useful anymore. I removed it in 7d572e9f6f09105c10c27d8ead1a1ee0f37f3c90. Also rebased the branch to current `develop`.
